### PR TITLE
pipeline created

### PR DESCRIPTION
**Pull request by kozae**

### CODE REVIEW

File: .ipynb_checkpoints/experiment_and_view-checkpoint.ipynb [hunk 1]
```diff
@@ -0,0 +1,11583 @@
+{
```
Comment (AbdBarho) [inline .ipynb_checkpoints/experiment_and_view-checkpoint.ipynb]: why is this file 12 000 lines long? maybe try committing the file after clearing the output of the cells

File: experiment_and_view.ipynb [hunk 2]
```diff
@@ -0,0 +1,2544 @@
+{
```
Comment (AbdBarho) [inline experiment_and_view.ipynb:1]: also this file, please clear the output

File: handlers/author_editor.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+def author_editor_cleanup(df):
+    # remove last column, its empty
+    df.drop(df.columns[len(df.columns) - 1], axis=1, inplace=True)
+    # replace all \N with NaN (pandas like it this way)
+    df.replace("\\N", np.nan, inplace=True)
+    # some authors names have a : at the end, remove that
+    df['authors'] = df['authors'].astype(str).apply(RemoveColon)
+    df['editors'] = df['editors'].astype(str).apply(RemoveColon)
+    return df
```
Comment (AbdBarho) [inline handlers/author_editor.py:12]: I think you missed the branch of @julianewundermann in which she worked on the clean up of the authors and editors.

see here: https://github.com/code-openness/Data/tree/7f933c45722947e7cbe97533a272c6379876f4cf

File: handlers/doi.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        else:
+            return value
+    else:
+        return value
+
+
+# a function to remove the value if not a valid DOI
+def remove_invalid_doi(value):
+    if type(value) is str:
+        doi_re = re.compile(r"10.\d{4,9}/[-._;()/:A-Z0-9]+$", re.I)
```
Comment (AbdBarho) [inline handlers/doi.py]: nice regex, no need to compile every time you call the function, you can compile it once outside it.

File: raw/export_to_sparql.ipynb [hunk 5]
```diff
@@ -1,561 +0,0 @@
-{
```
Comment (AbdBarho) [inline raw/export_to_sparql.ipynb]: why was this file deleted?

File: utils/apply_cleanup.py [hunk 6]
```diff
@@ -0,0 +1,4 @@
+def apply_cleanup(pik_df, functions):
+    for function in functions:
+        pik_df = function(pik_df)
+    return pik_df
```
Comment (AbdBarho) [inline utils/apply_cleanup.py:4]: there is no need to create a module for 4 lines of code, but its okay...

**Review by AbdBarho — CHANGES_REQUESTED** [review]
- I like the way how you created the pipeline, perfect!
- pik_output.csv should be the result of executing the scripts and thus should not exist

I have wrote some additional comments that hopefully address the problems

File: handlers/doi.py (hunk 4 above)
Comment (kozae) [reply handlers/doi.py]: Good catch!

File: .ipynb_checkpoints/experiment_and_view-checkpoint.ipynb (hunk 1 above)
Comment (kozae) [reply .ipynb_checkpoints/experiment_and_view-checkpoint.ipynb]: That is the checkpoints file, I am just going to delete it.

File: experiment_and_view.ipynb (hunk 2 above)
Comment (kozae) [reply experiment_and_view.ipynb:1]: Done!

File: raw/export_to_sparql.ipynb (hunk 5 above)
Comment (kozae) [reply raw/export_to_sparql.ipynb]: I thought it, it does not necessarily belong to the cleanup pipeline. It is a totally different concern, i.e. constructing the data-model and generating meta-data for SPARQL import, as far as I understand. If we decide to make the cleanup pipeline and the SPARQL export pipeline into a single entity I will work on finding a place in the workflow.

File: utils/apply_cleanup.py (hunk 6 above)
Comment (kozae) [reply utils/apply_cleanup.py:4]: I find separation of concerns in code structure is always needed, no matter how few code lines are . This function is also expected to get bigger as I add error handling procedures.

File: handlers/author_editor.py (hunk 3 above)
Comment (kozae) [reply handlers/author_editor.py:12]: OK, I was also wondering about that, the code is in "find_erroneous_data.ipynb", I will look into it. This may take a bit of time.

Comment (AbdBarho) [reply handlers/author_editor.py:12]: I will merge this pull request, you can then create another one when you have the code ready
